tobler/area_weighted/area_interpolate.py

```diff
@@ -9,10 +9,135 @@
 import warnings
 from scipy.sparse import dok_matrix, diags, coo_matrix
 import pandas as pd
+import os
 
 from tobler.util.util import _check_crs, _nan_check, _inf_check, _check_presence_of_crs
 
 
+def _chunk_dfs(geoms_to_chunk, geoms_full, n_jobs):
+    chunk_size = np.int64(geoms_to_chunk.shape[0] / n_jobs) + 1
+    for i in range(n_jobs):
+        start = i * chunk_size
+        yield geoms_to_chunk.iloc[start : start + chunk_size], geoms_full
+
+
+def _index_n_query(geoms1, geoms2):
+    # Pick largest for STRTree, query_bulk the smallest
+    if geoms1.shape[0] > geoms2.shape[0]:
+        large = geoms1
+        small = geoms2
+    else:
+        large = geoms2
+        small = geoms1
+    # Build tree + query
+    qry_polyIDs, tree_polyIDs = large.sindex.query_bulk(small, predicate="intersects")
+    # Remap IDs to global
+    large_global_ids = large.iloc[tree_polyIDs].index.values
+    small_global_ids = small.iloc[qry_polyIDs].index.values
+    # Return always global IDs for geoms1, geoms2
+    if geoms1.shape[0] > geoms2.shape[0]:
+        return np.array([large_global_ids, small_global_ids]).T
+    else:
+        return np.array([small_global_ids, large_global_ids]).T
+
+
+def _chunk_polys(id_pairs, geoms_left, geoms_right, n_jobs):
+    chunk_size = np.int64(id_pairs.shape[0] / n_jobs) + 1
+    for i in range(n_jobs):
+        start = i * chunk_size
+        chunk1 = geoms_left.values.data[id_pairs[start : start + chunk_size, 0]]
+        chunk2 = geoms_right.values.data[id_pairs[start : start + chunk_size, 1]]
+        yield chunk1, chunk2
+
+
+def _intersect_area_on_chunk(geoms1, geoms2):
+    import pygeos
+
+    intersection = pygeos.intersection(geoms1, geoms2)
+    areas = pygeos.measurement.area(intersection)
```

Comment:
```suggestion
    intersection = geoms1.intersection(geoms2)
    areas = intersection.area
```

`geoms1` and `geoms1` are not Numpy arrays of pygeos geometries here, are they? If I am reading it correctly it is `GeometryArray` or `GeoSeries`, so the code above should work as well without a direct import of pygeos. It also makes it work with pure shapely without pygeos (not that one would want to do that).

Reply:
Great point!